docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+These assumptions are: (i) partial synchrony is necessary for achieving liveness of a chain if that chain is implemented using a deterministic consensus protocol (e.g., Tendermint Core); (ii) synchrony is necessary for guaranteeing message delivery in any connection-oriented protocol such as TCP, which is most likely the underlying communication layer running between `I` and `R`.
+
+
+#### Properties
+
+###### Preliminary definitions
+We introduce two abstractions that are relevant at this point:
+
+1. A `ConnectionEnd` is a data type (i.e., interface) that abstracts over the details of a connection at a process.
+One of the goals of the connection handshake protocol is to  allocate an object of this type at each of the initiator and responder process.
+
+2. A `State` is a data type that abstracts over the details of the state of a process at a current point in time (e.g., a summary or snapshot).
+In the context of the CH problem, a process uses this data type to (i) declare their state to another process, and (ii) represent the state of another process.
+
+Knowing the precise declaration of `ConnectionEnd` and `State` is not necessary in understanding the properties of the CH protocol.
+Suffice to say that each `ConnectionEnd` has associated the `State` of both the initiator and responder process.
+We will declare this data types later (see [2.2 Protocol](#22-protocol)).
+
+We now restate the connection handshake problem in a slightly more precise (or lower level) formulation:
+A connection handshake enables two processes to agree on a consistent view of their state, to verify each other's state, and to allocate a new connection.
+
+
+###### Guarantees
+We refine the safety guarantee **[CH/Seq/1/Agreement]**  (defined in the [sequential problem statement](#sequential-problem-statement)) by splitting this into four complementary safety properties:
+
+- **[CH/Proto/1/Uniqueness]**
+If a process finishes a handshake for a `ConnectionEnd` `c`, no handshake has previously finished for `c`.
+
+- **[CH/Proto/2/Validity]**
+If either of the responder or initiator agree to a `ConnectionEnd` `c`, then the initiator proposed `c`.
+
+- **[CH/Proto/3/StateConsistency]**
+If process `i` establishes a connection with some process `r`, then process `i` has the `State` `si` representing process `r` and this state is consistent with the actual state of `r`.
+
+- **[CH/Proto/4/StateVerification]**
+If the processes establish a connection  for their respective `State` `si` (state of initiator) and `sr` (state of responder), then the initiator verified `sr` and the responder verified `si`.
+
+The liveness property below is functionally identical to the **[CH/Seq/2/Termination]** property we defined earlier:
+
+- **[CH/Proto/5/Termination]** If the initiator and responder are correct, then the handshake eventually terminates (i.e., both processes allocate a new `ConnectionEnd` object).
```

Comment:
should we say `correct termination` and require consistent `ConnectionEnd` objects?

Reply:
These properties are only intended to cover the case when both initiator and responder are correct, and in such a case, there is no possibility of incorrect/error termination. But you're raising a good point: what does it mean for a connection handshake to terminate incorrectly? Should we capture this in the specification?